### PR TITLE
feat(note-history): added note-history page

### PR DESCRIPTION
**Pull request by e11sy**
image

- added note history page
- added NoteHistory domain and application services
- added noteHistory repository
- added parseDate util

### Todo
- [x] add note title update for heading and for tab in tabbar
- [x] wait for codex icon 2.0 (icon `eye` needed) (left @todo for that)

Will make note history button in note page and note history record clickable in next PR

### CODE REVIEW

File: src/infrastructure/utils/parseDate.ts [hunk 1]
```diff
@@ -0,0 +1,19 @@
+export function parseDate(date: Date): string {
+  // Массив с названиями месяцев
+  const months = ['Jan', 'Feb', 'Mar', 'Apr', 'May', 'Jun', 'Jul', 'Aug', 'Sep', 'Oct', 'Nov', 'Dec'];
```
Comment (neSpecc) [inline src/infrastructure/utils/parseDate.ts:3]: we can try using native Intl module https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Intl/DateTimeFormat

File: src/presentation/pages/History.vue [hunk 2]
```diff
@@ -0,0 +1,139 @@
...
+            <template #left>
+              <Avatar
+                :src="historyRecord.user.photo"
+                :username="historyRecord.user.name"
+              />
+            </template>
+            <template #right>
+              <Button
+                secondary
+                icon="Search"
```
Comment (neSpecc) [inline src/presentation/pages/History.vue:40]: there is no icon in figma

File: src/presentation/pages/History.vue [hunk 3]
```diff
@@ -0,0 +1,139 @@
...
+                {{ t('history.view') }}
+              </Button>
+            </template>
+          </Row>
+        </Container>
+      </div>
+    </div>
+  </ThreeColsLayout>
+</template>
+<!-- @todo Make button with icon Eye, it would be implemented in Codex icons 2.0 -->
```
Comment (neSpecc) [inline src/presentation/pages/History.vue:51]: what is this?

File: src/presentation/pages/History.vue (hunk 2 above)
Comment (e11sy) [reply src/presentation/pages/History.vue:40]: image
this is temporary solution for Eye icon (as i can see it is not implemented in current version of codex icons)

Comment (e11sy) [reply src/presentation/pages/History.vue]: removed

**Review by neSpecc — APPROVED** [review]

**Review by TatianaFomina — APPROVED** [review]

**Pull request merged**